### PR TITLE
Update O# integration tests to use jest

### PR DESCRIPTION
**Pull request by dibarbet**
Also removes the remaining parts of mocha 🥳

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -44,224 +44,60 @@
...
-            "name": "Omnisharp: Launch Feature Tests",
+            "name": "Omnisharp: Launch Current File Integration Tests",
             "type": "extensionHost",
             "request": "launch",
             "runtimeExecutable": "${execPath}",
             "args": [
                 // Create a temp profile that has no extensions / user settings.
                 // This allows us to only have the C# extension + the dotnet runtime installer extension dependency.
                 "--profile-temp",
+                "${workspaceRoot}/omnisharptest/omnisharpIntegrationTests/testAssets/${input:omnisharpAssetName}/.vscode/omnisharp_${input:omnisharpAssetName}.code-workspace",
```
Comment (dibarbet) [inline .vscode/launch.json:55]: removed all the configs for the various asset files.  Now when this is selected, we show a quick pick to choose which assets to use.

File: .vscode/launch.json [hunk 2]
```diff
@@ -312,28 +148,43 @@
...
+            "program": "${workspaceFolder}/node_modules/gulp/bin/gulp.js",
+            "args": [
+                "test:razorintegration"
+            ],
+            "cwd": "${workspaceFolder}"
+        }
+    ],
+    "inputs": [
+        {
+            "id": "omnisharpAssetName",
```
Comment (dibarbet) [inline .vscode/launch.json:177]: this is the new quick pick definition

File: omnisharptest/omnisharpIntegrationTests/testAssets/slnWithCsproj/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+**/obj/
```
Comment (dibarbet) [inline omnisharptest/omnisharpIntegrationTests/testAssets/slnWithCsproj/.gitignore:1]: this is moved

I decided it was getting to be more annoying that it was worth to keep re-using the exact same assets for our tools host and O#.  By duplicating them, we can iterate on the new tests without worrying about breaking O#.

File: package.json [hunk 4]
```diff
@@ -117,15 +115,10 @@
   "devDependencies": {
     "@jest/globals": "^29.6.2",
     "@types/archiver": "5.1.0",
-    "@types/chai": "4.2.16",
```
Comment (dibarbet) [inline package.json:120]: 🥳

File: tasks/testTasks.ts [hunk 5]
```diff
@@ -3,44 +3,17 @@
...
-} from './projectPaths';
+import { codeExtensionPath, rootPath, outPath } from './projectPaths';
 import spawnNode from './spawnNode';
 import * as jest from 'jest';
 import { Config } from '@jest/types';
 import { jestOmniSharpUnitTestProjectName } from '../omnisharptest/omnisharpUnitTests/jest.config';
 import { jestUnitTestProjectName } from '../test/unitTests/jest.config';
 import { razorTestProjectName } from '../test/razorTests/jest.config';
-
-gulp.task('omnisharptest:feature', async () => {
```
Comment (dibarbet) [inline tasks/testTasks.ts:26]: these were turned into unit tests.  no need for them to run like this.

File: tasks/testTasks.ts [hunk 6]
```diff
@@ -109,53 +83,67 @@ gulp.task(
...
-        EXTENSIONS_TESTS_PATH: path.join(codeTestsPath, 'index.js'),
-    };
+/**
+ * Runs jest based integration tests.
+ * @param testAssetName the name of the test asset
+ * @param testFolderName the relative path (from workspace root)
+ * @param workspaceFileName the name of the vscode workspace file to use.
+ * @param env any environment variables needed.
+ */
+async function runJestIntegrationTest(
```
Comment (dibarbet) [inline tasks/testTasks.ts:115]: slightly interesting to review

File: test/vscodeLauncher.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as cp from 'child_process';
```
Comment (dibarbet) [inline test/vscodeLauncher.ts:6]: this was split from the old integration test runner.  We actually only need 1 copy of this, and not one for each jest integration test project (cc @davidwengier ).  But we do still need 1 index per jest project

File: .vscode/launch.json (hunk 1 above)
Comment (dibarbet) [reply .vscode/launch.json:55]: image

**Review by davidwengier — APPROVED** [review]
If the tests pass, who am I to judge :)

**Review by ryzngard — APPROVED** [review]

**Review by akhera99 — APPROVED** [review]

**Pull request merged**